src/Dapr.Actors/Communication/ActorCommunicationExceptionData.cs

```diff
@@ -0,0 +1,73 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors.Communication
+{
+    using System;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Xml;
+    using Dapr.Actors;
+    using Microsoft.Extensions.Logging;
+
+    [DataContract(Name = "ActorCommunicationExceptionData", Namespace = Constants.Namespace)]
+    internal class ActorCommunicationExceptionData
+    {
+        private static readonly DataContractSerializer ActorCommunicationExceptionDataSerializer = new DataContractSerializer(typeof(ActorCommunicationExceptionData));
+
+        public ActorCommunicationExceptionData(string type, string message)
+        {
+            this.Type = type;
+            this.Message = message;
+        }
+
+        [DataMember]
+        public string Type { get; private set; }
+
+        [DataMember]
+        public string Message { get; private set; }
+
+        internal static bool TryDeserialize(Stream data, out ActorCommunicationExceptionData result, ILogger logger = null)
+        {
+            try
+            {
+                var exceptionData = Deserialize(data);
+                result = exceptionData;
+                return true;
+            }
+            catch (Exception e)
+            {
+                // swallowing the exception
+                logger?.LogWarning(
+                    "ActorCommunicationException",
+                    " ActorCommunicationExceptionData DeSerialization failed : Reason  {0}",
+                    e);
+            }
+
+            result = null;
+            return false;
+        }
+
+        internal static ActorCommunicationExceptionData Deserialize(Stream buffer)
+        {
+            if ((buffer == null) || (buffer.Length == 0))
+            {
+                return null;
+            }
+
+            using var reader = XmlDictionaryReader.CreateBinaryReader(buffer, XmlDictionaryReaderQuotas.Max);
+            return (ActorCommunicationExceptionData)ActorCommunicationExceptionDataSerializer.ReadObject(reader);
+        }
+
+        internal byte[] Serialize()
+        {
+            using var stream = new MemoryStream();
+            using var writer = XmlDictionaryWriter.CreateBinaryWriter(stream);
+            ActorCommunicationExceptionDataSerializer.WriteObject(writer, this);
+            writer.Flush();
+            return stream.ToArray();
```

Comment:
Why this serialization format? This is a binary-encoded XML format - why not something well known like JSON?

I'm not sure whether or not this is suitable for untrusted data - It's not an API I am familar with

Reply:
I'm fine with decision as long as we plan to address it before 1.0. 

Beyond issues like security, and configuration - there's more at stake here as well - cost to load and JIT additional types, memory consumption from the same, and binary size for anyone doing a trimmed publish.